vendor/bat-native-ads/src/bat/ads/internal/processors/behavioral/purchase_intent/purchase_intent_processor.cc

```diff
@@ -224,7 +224,7 @@ void PurchaseIntent::OnResourceDidUpdate(const std::string& id) {
 }
 
 void PurchaseIntent::OnTextContentDidChange(
-    const int32_t id,
+    const int32_t tab_id,
     const std::vector<GURL>& redirect_chain,
     const std::string& content) {
   const GURL& url = redirect_chain.back();
```

Comment:
we need to check redirect_chain size before calling back() method

Reply:
The redirect chain is always guaranteed to have at least 1 entry. See `navigation_handle->GetRedirectChain()`

Reply:
in this case we need to add DCHECK here

Reply:
Added, thanks